.github/workflows/ci_test.yml

```diff
@@ -3,34 +3,12 @@ on: [push, pull_request]
 name: CI test and lint
 
 jobs:
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout sources
-        uses: actions/checkout@v2
-
-      - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-
-      - name: Run cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-
-      - name: Run cargo check nodefault
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-
-
   test:
     name: Test Suite
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
I know it's free to run them all, but this just feels like wasting resources. There is nothing os-specific in this library. What would make sense would be to run on different cpu architectures (something I planned to add but didn't get around to yet).

Reply:
macos at least is useful since it's a common platform and it doesn't have the fma CPU feature. Ultimately it's a bit of an issue that CI isn't able to at least perform a positive test on hardware configuration (e.g. neon is not covered) which can easily let errors slip by.

You want me to nuke this?

Reply:
With the free github actions you don't know what cpu the job will run on. They have a very wide range of machines of different age, and it's really a lottery what you get. I wouldn't think all macos hosts lack fma (and I also wouldn't be sure all linux hosts have it). 

About arm and neon, that is possible to test with cross: https://github.com/ejmahler/RustFFT/blob/master/.github/workflows/run_test.yml#L63-L119
Those two tests are quite useful. 32-bit x86 checks that everything works as expected on a 32-bit machine, and since it runs without emulation it's still fast. And aarch64 can test with neon. That runs with qemu, so it's pretty slow. The tests here run pretty quickly so it probably won't be an issue. But if they take too long the tests can run be run with --release.

At some point there should also be tests running on several rustc versions. Let's drop windows and macos, to not make the list of runs overly long.

Reply:
Done